src/monmove.cpp

```diff
@@ -341,7 +342,10 @@ void monster::plan( const mfactions &factions )
     } else if( friendly < 0 && sees( g->u ) ) {
         if( rl_dist( pos(), g->u.pos() ) > 2 ) {
             set_dest( g->u.pos() );
-        } else {
+        }
+        else if( has_flag( MF_EGGLAYING ) && has_effect( effect_laid_egg ) ) {
```

Comment:
If it's called unconditionally and should do nothing for most monsters how do you check which monster to apply it?